speechbrain/utils/distributed.py

```diff
@@ -12,9 +13,62 @@
 
 import torch
 
+from speechbrain.utils.distributed_utils import recursively_apply
+
 MAIN_PROC_ONLY = 0
 
 
+class DistributedState:
+    """
+    Singleton class that has information about the current training environment and functions to help with process
+    control.
+    """
+
+    _shared_state = dict()
+
+    def __init__(self, device: str = "cpu", distributed_backend=None):
```

Comment:
I think the singleton pattern in python typically uses `__new__` rather than `__init__`

Reply:
I removed the class DistributedState since I agree that torch.distributed is already providing the needed features. At the beginning I was thinking about using this class for the future if we want to support other backend such as TPU, Megatron etc but I think it's not worth it to do it now. (the idea was to abstract some key functions that can differ depending on the backend used).